frontends/web/src/routes/device/bitbox02/createbackup.tsx

```diff
@@ -15,83 +15,66 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
-import { createBackup } from '../../../api/bitbox02';
-import { translate, TranslateProps } from '../../../decorators/translate';
-import { apiPost } from '../../../utils/request';
+import { useState } from 'react';
+import { checkBackup, createBackup as createBackupAPI } from '../../../api/bitbox02';
 import { alertUser } from '../../../components/alert/Alert';
 import { confirmation } from '../../../components/confirm/Confirm';
 import { Button } from '../../../components/forms';
 import { WaitDialog } from '../../../components/wait-dialog/wait-dialog';
+import { useTranslation } from 'react-i18next';
 
-interface CreateProps {
+type TProps = {
     deviceID: string;
 }
 
-type Props = CreateProps & TranslateProps;
+export const Create = ({ deviceID }: TProps) => {
+  const [creatingBackup, setCreatingBackup] = useState(false);
+  const [disabled, setDisabled] = useState(false);
+  const { t } = useTranslation();
 
-interface State {
-    creatingBackup: boolean;
-    disabled: boolean;
-}
-
-class Create extends Component<Props, State> {
-  public readonly state: State = {
-    creatingBackup: false,
-    disabled: false,
-  };
-
-  private createBackup = () => {
-    this.setState({ creatingBackup: true });
-    createBackup(this.props.deviceID)
-      .then(() => this.setState({ creatingBackup: false, disabled: false }))
+  const createBackup = () => {
+    setCreatingBackup(true);
+    createBackupAPI(deviceID)
+      .then(() => {
+        setCreatingBackup(false);
+        setDisabled(false);
+      })
       .catch(() => {
-        alertUser(this.props.t('backup.create.fail'));
+        setCreatingBackup(false);
+        setDisabled(false);
+        alertUser(t('backup.create.fail'));
       });
   };
 
-  private maybeCreateBackup = () => {
-    this.setState({ disabled: true });
-    apiPost('devices/bitbox02/' + this.props.deviceID + '/backups/check', {
-      silent: true,
-    }).then(check => {
-      if (check.success) {
-        confirmation(this.props.t('backup.create.alreadyExists'), result => {
-          if (result) {
-            this.createBackup();
-          } else {
-            this.setState({ disabled: false });
-          }
-        });
-        return;
-      }
-      this.createBackup();
-    });
+  const maybeCreateBackup = async () => {
+    setDisabled(true);
+    const check = await checkBackup(deviceID, true);
+    if (check) {
+      confirmation(t('backup.create.alreadyExists'), result => {
+        if (result) {
+          createBackup();
+        } else {
+          setDisabled(false);
+        }
+      });
+      return;
+    }
+    createBackup();
```

Comment:
Testing create-backup failed on empty sdCard / nothing happened in the UI, got uncaught promise error.

This is another reason why checkBackup shouldn't throw an error if it does not find a backup. One solution here for now could be to wrap the whole thing in try/catch and do the createBackup outside, or add .catch(console.error);

Reply:
sorry i guess this wouldn't have happened if I didn't advice you to use checkBackup from api/bitbox02

Reply:
```diff
diff --git a/frontends/web/src/routes/device/bitbox02/createbackup.tsx b/frontends/web/src/routes/device/bitbox02/createbackup.tsx
index 11edd5d9d..99936b4dd 100644
--- a/frontends/web/src/routes/device/bitbox02/createbackup.tsx
+++ b/frontends/web/src/routes/device/bitbox02/createbackup.tsx
@@ -48,7 +48,7 @@ export const Create = ({ deviceID }: TProps) => {
 
   const maybeCreateBackup = async () => {
     setDisabled(true);
-    const check = await checkBackup(deviceID, true);
+    const check = await checkBackup(deviceID, true).catch(console.error);
     if (check) {
       confirmation(t('backup.create.alreadyExists'), result => {
         if (result) {
@@ -58,8 +58,9 @@ export const Create = ({ deviceID }: TProps) => {
         }
       });
       return;
+    } else {
+      createBackup();
     }
-    createBackup();
   };
 
   return (
```

Reply:
opened a new PR with this change: https://github.com/digitalbitbox/bitbox-wallet-app/pull/1985